src/ScriptBuilder/Saga/Writers/OracleSagaScriptWriter.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using NServiceBus.Persistence.Sql.ScriptBuilder;
+
+class OracleSagaScriptWriter : ISagaScriptWriter
+{
+    TextWriter writer;
+    SagaDefinition saga;
+    string tableName;
+
+    public OracleSagaScriptWriter(TextWriter textWriter, SagaDefinition saga)
+    {
+        writer = textWriter;
+        this.saga = saga;
+        if (saga.TableSuffix.Length > 27)
+        {
+            throw new Exception($"Saga '{saga.TableSuffix}' contains more than 27 characters, which is not supported by SQL persistence using Oracle. Either disable Oracle script generation using the SqlPersistenceSettings assembly attribute, shorten the name of the saga, or specify an alternate table name using the SqlSagaAttribute's 'tablePrefix' parameter.");
+        }
+        if (Encoding.UTF8.GetBytes(saga.TableSuffix).Length != saga.TableSuffix.Length)
```

Comment:
This can't be centralized because it's needed both in SqlPersistence and ScriptBuilder and they have no common dependency. Given that and it's not a hot path, isn't it better to have a simple one-liner rather than call out to another method? I also think it conveys intent better than `saga.TableSuffix.Any(ch => (int)ch > 127)`.